### ISSUE TITLE
Features that I am currently missing

### ISSUE THREAD

**Issue by mihhail-lapushkin** [ericdrowell/KineticJS#225]
Hi,

While working on a project using KineticJS I've collected a bunch of extensions that I've created for myself. Would be nice to include them into the library.
## Convenience methods for Containers

```
prototype.isEmpty = function() {
    return this.getChildren().length === 0;
};

prototype.size = function() {
    return this.getChildren().length;
};

prototype.each = function(fn) {
    this.getChildren().forEach(fn); // for library the loop might be better for compatibility reasons
};
```
## toLocal and toGlobal methods (like in ActionScript)

Here is my code for Container:

```
prototype._toCoord = function(args, global) {
    var pos = Kinetic.Type._getXY([].slice.call(args));
    var node = this;

    while (node.nodeType !== 'Stage') {
        pos.x += (global ? 1 : -1) * node.getX();
        pos.y += (global ? 1 : -1) * node.getY();
        node = node.getParent();
    }

    return pos;
};

prototype.toGlobal = function() {
    return this._toCoord(arguments, true);
};

prototype.toLocal = function() {
    return this._toCoord(arguments, false);
};
```

toGlobal can also be added to Nodes without arguments.
## Shortcuts for setListening

Like with setVisible, setListening can also have convenient shortcuts.
For example:
stopListening [-]
cancelListening [-]
supressListening [-]
dontListen [-]
listen [+]
## Image proportion sync

Add automatic image proportion sync on resize. Either as an on/off property of Image or as a separate class like I did:

```
Kinetic.ProportionalImage = function(config) {
    Kinetic.Image.call(this, config);

    this.on('widthChange', this._syncHeight);
    this.on('heightChange', this._syncWidth);

    if (config.width != null) {
        this.setWidth(config.width);
    } else if (config.height != null) {
        this.setHeight(config.height);
    }
};

Kinetic.ProportionalImage.prototype = {
    _syncDim: function(from, to, evt) {
        var img = this.getImage();

        this.attrs[to] = img[to] * evt.newVal / img[from];
    },

    _syncWidth: function(evt) {
        this._syncDim('height', 'width', evt);
    },

    _syncHeight: function(evt) {
        this._syncDim('width', 'height', evt);
    }
};

Kinetic.Global.extend(Kinetic.ProportionalImage, Kinetic.Image);
```
## 0 Opacity = hidden

Treat 0 opacity as a hidden Node i.e. don't draw it. Now I have to hide the Node after fading out to save those precious CPU ticks on useless drawing.
## Accelerated canvas environment support

As you maybe know currently some companies are making Accelerated canvases for mobile environments (directCanvas and CocoonJS). Those environments don't have all features of HTML and that's why KineticJS does not work out of the box.
I use CocoonJS and for KineticJS to work I've made following changed to Stage:

```
State.prototype._buildDOM = function() {
    this.content = this.attrs.container;

    this.bufferCanvas = new Kinetic.Canvas();
    this.hitCanvas = new Kinetic.Canvas(0, 0, true);

    this._resizeDOM();
};

State.prototype._getContentPosition = function() {
    var rect = this.content.getBoundingClientRect ? this.content.getBoundingClientRect() : { top : 0, left : 0 };
    return { top: rect.top, left: rect.left };
};
```

Basically what is needed is to:
- Check getBoundingClientRect actually exists and if not return 0,0.
- Add a configuration option to Stage constructor not to make the div wrapper for canvases, but to append them to container directly.
